### PR TITLE
Variables: Refreshes all panels even if panel is full screen

### LINKED ISSUE #17537: Variables: Filter is not applied to other panels if the variable value is changed while one panel is open in edit/view mode
**What happened**:
I have two similar tables, they use the same template variables, and the filters work as expected if the panels are closed. If I open one of the tables in edit/view mode, change the value of the filter, and go back to the dashboard, the filter is only applied to the table that was in open mode (until I manually refresh the dashboard).

**What you expected to happen**:
All the panels are refreshed when a filter is changed (regardless of the currently open panel). 

**How to reproduce it (as minimally and precisely as possible)**:
- Create a new dashboard
- Create a template variable
- Add two panels, that use the template variable,  to the dashboard 
- Open one of the panels in edit mode
- Change the filter value
- Go back to the panel

**Anything else we need to know?**:

**Environment**:
- Grafana version: 6.2.0
- Data source type & version: ES 5.6
- OS Grafana is installed on: Centos 7
- User OS & Browser: Centos 7, Firefox 60.6.1esr (64-bit)

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
This PR adds a function called `changeAffectsAllPanels` to the DashboardModel that can be called for changes that affect all panels, such as variable changes. When the panel then exits fullscreen view or edit view all panels will be refreshed.

**Which issue(s) this PR fixes**:

Fixes #17537

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/dashboard/state/DashboardModel.ts [hunk 1]
```diff
@@ -114,6 +115,9 @@ export class DashboardModel {
     panelInView: true,
     getVariablesFromState: true,
     formatDate: true,
+    hasChangesThatAffectsAllPanels: true,
+    changeAffectsAllPanels: true,
```
Comment (torkelo) [inline public/app/features/dashboard/state/DashboardModel.ts:119]: you do not need to add functions here, only value props

Comment (hugohaggmark) [reply public/app/features/dashboard/state/DashboardModel.ts:119]: Gotcha, thank you!

File: public/app/features/dashboard/state/DashboardModel.ts [hunk 2]
```diff
@@ -365,11 +370,30 @@ export class DashboardModel {
...
     this.panelInEdit = undefined;
+    this.refreshIfChangeAffectsAllPanels();
+  }
+
+  changeAffectsAllPanels() {
+    if (!this.panelInEdit && !this.panelInView) {
+      return;
+    }
+
+    this.hasChangesThatAffectsAllPanels = true;
```
Comment (torkelo) [inline public/app/features/dashboard/state/DashboardModel.ts:387]: maybe more intuitive to write this as:

```ts
if (this.panelInEdit || this.panelInView) {
   this.hasChangesThatAffectsAllPanels = true;
}
```
That way you do not have to invert the booleans in your to understand when this is set to true :)

or just  this.hasChangesThatAffectsAllPanels = this.panelInEdit || this.panelInView;

File: public/app/features/variables/state/actions.ts [hunk 3]
```diff
@@ -526,6 +527,7 @@ export const onTimeRangeUpdated = (
   try {
     await Promise.all(promises);
     const dashboard = getState().dashboard.getModel();
+    dashboard?.changeAffectsAllPanels();
     dashboard?.startRefresh();
```
Comment (torkelo) [inline public/app/features/variables/state/actions.ts:531]: wonder if this makes more sense as param to startRefresh ? like dashboard.startRefresh({ changeAffectsAllPanels: true }) ;

File: public/app/features/dashboard/state/DashboardModel.ts [hunk 4]
```diff
@@ -365,11 +370,30 @@ export class DashboardModel {
...
+    this.refreshIfChangeAffectsAllPanels();
   }
 
   exitPanelEditor() {
     this.panelInEdit!.destroy();
     this.panelInEdit = undefined;
+    this.refreshIfChangeAffectsAllPanels();
+  }
+
+  changeAffectsAllPanels() {
```
Comment (torkelo) [inline public/app/features/dashboard/state/DashboardModel.ts:382]: this naming implies a boolean return value for me (ie reads like a question) not a state change action.

**Review by torkelo — APPROVED** [review]
Looks good, found a few nits and alt options

File: public/app/features/dashboard/state/DashboardModel.ts (hunk 2 above)
Comment (hugohaggmark) [reply public/app/features/dashboard/state/DashboardModel.ts:387]: I was thinking about that too, thanks!

File: public/app/features/variables/state/actions.ts (hunk 3 above)
Comment (hugohaggmark) [reply public/app/features/variables/state/actions.ts:531]: Ok, so then the caller is responsible for checking isPanelInEdit || isPanelInView? I wanted to separate those?

File: public/app/features/dashboard/state/DashboardModel.ts (hunk 4 above)
Comment (hugohaggmark) [reply public/app/features/dashboard/state/DashboardModel.ts:382]: Yeah, naming I'll try to figure something out.

File: public/app/features/dashboard/state/DashboardModel.ts (hunk 2 above)
Comment (hugohaggmark) [reply public/app/features/dashboard/state/DashboardModel.ts]: Fixed!

File: public/app/features/dashboard/state/DashboardModel.ts (hunk 4 above)
Comment (hugohaggmark) [reply public/app/features/dashboard/state/DashboardModel.ts]: Keeping this for now.

File: public/app/features/variables/state/actions.ts (hunk 3 above)
Comment (hugohaggmark) [reply public/app/features/variables/state/actions.ts:531]: Keeping this for now

File: public/app/features/dashboard/state/DashboardModel.ts (hunk 4 above)
Comment (hugohaggmark) [reply public/app/features/dashboard/state/DashboardModel.ts]: Fixed

**Pull request merged**